qiskit/providers/aer/pulse_new/models/quantum_models.py

```diff
@@ -0,0 +1,245 @@
+                            operator is diagonal
+
+        Returns:
+            np.array: the evaluated model
+        """
+
+        if self.signals is None:
+            raise Exception("""OperatorModel cannot be
+                               evaluated without signals.""")
+
+        sig_vals = self.signals.value(time)
+
+        op_combo = self._evaluate_in_frame_basis_with_cutoffs(sig_vals)
+
+        op_to_add_in_fb = None
+        if self.frame.frame_operator is not None:
+            op_to_add_in_fb = -1j * np.diag(self.frame.frame_diag)
+
+
+        return self.frame._conjugate_and_add(time,
+                                             op_combo,
+                                             op_to_add_in_fb=op_to_add_in_fb,
+                                             operator_in_frame_basis=True,
+                                             return_in_frame_basis=in_frame_basis)
+
+class QuantumSystemModel:
+    """A model of a quantum system, consisting of a :class:`HamiltonianModel`
+    and an optional description of dissipative dynamics.
+
+    Dissipation terms are understood in terms of the Lindblad master
+    equation:
+
+    .. math::
+        \dot{\rho}(t) = -i[H(t), \rho(t)] + \mathcal{D}(t)(\rho(t)),
+
+    where :math:`\mathcal{D}(t)` is the dissipator portion of the equation,
+    given by
+
+    .. math::
+        \mathcal{D}(t)(\rho(t)) = \sum_j \gamma_j(t) L_j \rho L_j^\dagger - \frac{1}{2} \{L_j^\dagger L_j, \rho\},
+
+    with :math:`[\cdot, \cdot]` and :math:`\{\cdot, \cdot\}` the
+    matrix commutator and anti-commutator, respectively. In the above:
+        - :math:`H(t)` denotes the Hamiltonian,
+        - :math:`L_j` denotes the :math:`j^{th}` noise, or Lindblad,
+          operator, and
+        - :math:`\gamma_j(t)` denotes the signal corresponding to the
+          :math:`j^{th}` Lindblad operator.
+    """
+
+    def __init__(self,
+                 hamiltonian: HamiltonianModel,
+                 noise_operators: Optional[List[Operator]] = None,
+                 noise_signals: Optional[Union[List[BaseSignal], VectorSignal]] = None):
+        """Initialize. Noise parameters are optional. If `noise_operators`
+        is specified but `noise_signals` is left as `None`, then internally
+        sets the coefficient for each noise operator to `Constant(1.)`.
+
+        Args:
+            hamiltonian: the Hamiltonian.
+            noise_operators: list of dissipation operators.
+            noise_signals: list of time-dependent signals for the dissipation
+                           operators.
+        """
+
+        self.hamiltonian = hamiltonian
+
+        self.noise_operators = noise_operators
+
+        if noise_signals is None and noise_operators is not None:
+            noise_signals = [Constant(1.) for _ in noise_operators]
+
+        self.noise_signals = noise_signals
+
+    @property
+    def vectorized_lindblad_generator(self):
```

Comment:
I was thinking the `QuantumSystemModel` may have more functionality beyond just representing the Lindblad equation - e.g. it could also construct the DE required for monte carlo simulation.

Reply:
Resolving this as the implications of this comment are addressed byt he previous one.